### PR TITLE
Do not break when binding databases with no tables

### PR DESCRIPTION
**Pull request by riccardomc**
This PR introduces a simple fix that prevents the service broker to fail when binding/unbinding databases with no tables.

When a service instance is first created there are no tables in the relative database. The following statements:
```
"GRANT ALL ON TABLE %s.* TO %s", dbName, user
"REVOKE ALL ON TABLE %s.* FROM %s", dbName, user
```
cause the service binding/unbingind to fail with the error `pq: no object matched`.

### CODE REVIEW

File: plans.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ type Plan struct {
 	CRDBPort      string `json:"crdbPort"`
 	CRDBAdminUser string `json:"crdbAdminUser"`
 
-	crdb        *sql.DB
+	crdb *sql.DB
```
Comment (a-robinson) [inline plans.go]: If this file only has whitespace changes in it, let's leave it out of the commit.

File: broker.go [hunk 2]
```diff
@@ -133,12 +134,17 @@ func (sb *crdbServiceBroker) Bind(
 	if _, err := plan.crdb.Exec(
 		fmt.Sprintf("GRANT ALL ON TABLE %s.* TO %s", dbName, user),
 	); err != nil {
-		cleanup()
```
Comment (a-robinson) [inline broker.go:136]: Don't we still need to clean up if the error matches `dbNotFoundErrRegexp`?

File: broker.go [hunk 3]
```diff
@@ -171,7 +177,10 @@ func (sb *crdbServiceBroker) Unbind(
 
 	if _, err := plan.crdb.Exec(fmt.Sprintf("REVOKE ALL ON TABLE %s.* FROM %s", dbName, user)); err != nil {
 		log.Error("revoke-grants", err)
-		return fmt.Errorf("remoking grants from tables for user: %s", err)
+		// if there are no tables in the database, we don't want to break
+		if !strings.Contains(err.Error(), "pq: no object matched") {
+			return fmt.Errorf("remoking grants from tables for user: %s", err)
```
Comment (a-robinson) [inline broker.go]: "revoking", not "remoking"

**Review by a-robinson — COMMENTED** [review]
cc @RaduBerinde in case there's a better way to do this.

File: broker.go [hunk 4]
```diff
@@ -133,12 +134,17 @@ func (sb *crdbServiceBroker) Bind(
...
 		fmt.Sprintf("GRANT ALL ON TABLE %s.* TO %s", dbName, user),
 	); err != nil {
-		cleanup()
 		if dbNotFoundErrRegexp.MatchString(err.Error()) {
 			return brokerapi.Binding{}, brokerapi.ErrInstanceDoesNotExist
 		}
 		log.Error("grant-privileges", err)
-		return brokerapi.Binding{}, fmt.Errorf("granting privileges: %s", err)
+
+		// if there are no tables in the database we don't want to clean and
```
Comment (RaduBerinde) [inline broker.go]: clean up? can also remove it and just say "we don't want to fail the binding"

File: broker.go [hunk 5]
```diff
@@ -133,12 +134,17 @@ func (sb *crdbServiceBroker) Bind(
...
+		// fail the binding
+		if !strings.Contains(err.Error(), "pq: no object matched") {
```
Comment (RaduBerinde) [inline broker.go]: [nit] I would define this as a regexp alongside the other two

**Review by RaduBerinde — COMMENTED** [review]
Thanks for the fix!

I looked around for a bit and can't think of a cleaner way to do it.

File: broker.go (hunk 2 above)
Comment (riccardomc) [reply broker.go:136]: yes, fixed: https://github.com/cockroachdb/pcf-crdb-service-broker/pull/6/files#diff-019034382244c82f15c92490660e4a18R138

File: broker.go (hunk 5 above)
Comment (riccardomc) [reply broker.go]: I thought about it, but it is just a string matching. In the other cases you actually have a pattern part. I've now introduced the regex for consistency.

**Review by a-robinson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**